README.md

```diff
@@ -126,6 +112,167 @@ func main() {
 	}
 }
 ```
-Nirvana server has a plugin machanism. All plugins are registered into config. The server will install them when the server is starting.
-`nirvana.Descriptor()` is a plugin method for installing API descriptors into config. After server starting, you could test echo API by `http://localhost:8080/echo?msg=test`.
 
+Now run the server and test it:
+
+```
+go run ./examples/getting-started/basics/echo.go
+INFO  0202-16:34:38.663+08 echo.go:65 | Listening on :8080
+INFO  0202-16:34:38.663+08 builder.go:163 | Definitions: 1 Middlewares: 0 Path: /echo
+INFO  0202-16:34:38.663+08 builder.go:178 |   Method: Get Consumes: [*/*] Produces: [text/plain]
+```
+
+In another terminal:
+
+```
+$ curl "http://localhost:8080/echo?msg=test"
+test
+```
+
+For full example code, see [basics](./examples/getting-started/basics).
+
+### Validate it!
+
+Now you are tired of echoing non-sense testing message and want to only reply message longer than 10 characters, such
+validation can be easily added when defining your descriptor:
+
+```go
+Parameters: []definition.Parameter{
+	{
+		Source:      definition.Query,
+		Name:        "msg",
+		Description: "Corresponding to the second parameter",
+		Operators:   []definition.Operator{validator.String("gt=10")},
+	},
+},
+```
+
+`Operator` is a concept in Nirvana to allow framework user to operate on input request; validation is one of several
+pre-defined operators. Another example of `operator` is `convertor`, which allows user to convert between different
+versions of an input.
+
+Under the hood, Nirvana uses [go-playground/validator.v9](https://github.com/go-playground/validator) for validation.
+which defines a list of useful tags, and supports custom validation. Nirvana integrates nicely with validator.v9
+package, see user guide for more advanced usage.
+
+Now run our new echo server and verify validation works:
+
+```
+$ go run echo.go
+INFO  0202-11:18:50.235+08 echo.go:67 | Listening on :8080
+INFO  0202-11:18:50.235+08 builder.go:163 | Definitions: 1 Middlewares: 0 Path: /echo
+INFO  0202-11:18:50.235+08 builder.go:178 |   Method: Get Consumes: [*/*] Produces: [text/plain]
+```
+
+In another terminal:
+
+```
+$ curl "http://localhost:8080/echo?msg=test"
+Key: '' Error:Field validation for '' failed on the 'gt' tag
```

Comment:
The error message is unable to understand.

Reply:
filed issue: https://github.com/caicloud/nirvana/issues/126